### PR TITLE
Fix #4840: Alias class prototype for methods in loose mode

### LINKED ISSUE #4840: [Feature Request] enable even simpler transformation
Please ignore if this already discussed.
Suppose I have a class like this:
```js
class MyClass {
   //....
   // Here I have 100 class methods
}
```

This will be transformed into:

```js
function MyClass() {}
// 100 times of this repeated: MyClass.prototype
```

Can we have something like this instead:
```js
var _proto = MyClass.prototype
// 1000 class methods ?
_proto.method1 = () => null;
_proto.method2 = () => null;

// a lot less output code after minification
// var a = MyClass.prototype;
// a.method1 = () => null;
// a.method2 = () => null;
```

Imagine if we have 20 of classes like this, the amount of prototype repeated will be big, not counted the long class names

Can we have it enhanced in this way ?

### PR DESCRIPTION
**Pull request by oliverdon**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          |  no
| Major: Breaking Change?  | no
| Minor: New Feature?      | no
| Deprecations?            | no
| Spec Compliancy?         | yes?
| Tests Added/Pass?        | yes
| Fixed Tickets            | Fixes #4840 
| License                  | MIT
| Doc PR                   | no
| Dependency Changes       | no

Alias class prototype for methods in loose mode.

### CODE REVIEW

File: packages/babel-plugin-transform-es2015-classes/src/loose.js [hunk 1]
```diff
@@ -6,14 +6,33 @@ export default class LooseClassTransformer extends VanillaTransformer {
...
     this.isLoose = true;
+    this.aliasInserted = false;
+    this.methodAlias = "_proto";
+  }
+
+  _insertProtoAliasOnce() {
+    if (!this.aliasInserted) {
+      const classProto = t.memberExpression(this.classRef, t.identifier("prototype"));
+      const protoDeclaration = t.variableDeclaration("var", [
+        t.variableDeclarator(t.identifier(this.methodAlias), classProto),
```
Comment (jridgewell) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: This needs to be a unique identifier, not the known I'd `_proto`.

Comment (oliverdon) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: Thanks, that makes sense.

would replacing line 10: 
```Javascript
this.methodAlias = "_proto";
``` 
with 
```Javascript
this.methodAlias = this.path.scope.generateUidIdentifier("proto").name;
```

be a good way of achieving this?

Comment (jridgewell) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: Yup.

Comment (oliverdon) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: Made the change, and caught up changes from 7.0.

File: packages/babel-plugin-transform-es2015-classes/src/loose.js [hunk 2]
```diff
@@ -6,6 +6,25 @@ export default class LooseClassTransformer extends VanillaTransformer {
...
+  }
+
+  _insertProtoAliasOnce() {
+    if (!this.aliasInserted) {
+      const classProto = t.memberExpression(
+        this.classRef,
+        t.identifier("prototype"),
+      );
+      const protoDeclaration = t.variableDeclaration("var", [
+        t.variableDeclarator(t.identifier(this.methodAlias), classProto),
```
Comment (hzoo) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: If you are going to use generateUidIdentifier, then you should just the identifier part and not do the `.name` 😄 

```js
this.methodAlias = this.path.scope.generateUidIdentifier("proto")
 t.variableDeclarator(this.methodAlias, classProto),
```

Comment (oliverdon) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: Ah, good point 😅.

File: packages/babel-plugin-transform-es2015-classes/src/loose.js [hunk 3]
```diff
@@ -6,6 +6,25 @@ export default class LooseClassTransformer extends VanillaTransformer {
   constructor() {
     super(...arguments);
     this.isLoose = true;
+    this.aliasInserted = false;
+    this.methodAlias = this.path.scope.generateUidIdentifier("proto");
```
Comment (jridgewell) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: Nit: this uses a global name. We can instead test if `methodAlias` is set yet, and create it when needed.

Comment (oliverdon) [inline packages/babel-plugin-transform-es2015-classes/src/loose.js]: Good point, removed this from constructor.

**Comment by hzoo** [comment]
@oliverdon thanks for your efforts! awesome stuff.

^ aside: btw we should figure out plugin options (loose mode) for the repl

**Pull request merged**